### PR TITLE
Possible fix for #4081 and #2707

### PR DESCRIPTION
**Pull request by T3N4K4**
I tried to create a fix for #4081 and #2707 where _NET_WM_FULLSCREEN_MONITORS gets detected automatically instead of the version before where you have to set +openbox.

I guess there are two approaches. Either search for _NET_WM_FULLSCREEN_MONITORS in _NET_WM_SUPPORTED, or read _NET_WM_FULLSCREEN_MONITORS out directly with xf_GetWindowProperty().

### CODE REVIEW

**Comment by akallabeth** [comment]
@freerdp-bot test

File: client/X11/xf_window.c [hunk 1]
```diff
@@ -203,10 +206,16 @@ void xf_SetWindowFullscreen(xfContext* xfc, xfWindow* window, BOOL fullscreen)
...
 	}
 
-	/* Set the fullscreen state */
-	xf_SendClientEvent(xfc, window->handle, xfc->_NET_WM_STATE, 4,
-	                   fullscreen ? _NET_WM_STATE_ADD : _NET_WM_STATE_REMOVE,
-	                   xfc->_NET_WM_STATE_FULLSCREEN, 0, 0);
+	status = xf_GetWindowProperty(xfc, DefaultRootWindow(xfc->display),
+	                              xfc->_NET_WM_FULLSCREEN_MONITORS, 1, &nitems, &bytes, &prop);
+
+	if (status)
```
Comment (hardening) [inline client/X11/xf_window.c]: Shouldn't we also check the value of the property ?

Comment (T3N4K4) [inline client/X11/xf_window.c]: Do you mean by checking _NET_WM_FULLSCREEN_MONITORS with XInternAtom() if it is "None" ?

Comment (hardening) [inline client/X11/xf_window.c]: I was meaning, checking the value of the retrieved property (prop) but perhaps I'm missing something...

Comment (T3N4K4) [inline client/X11/xf_window.c]: Ah I understand. I will rework it next week on monday or @nfedera could check it.

I was a bit confused.
 _NET_WM_FULLSCREEN_MONITORS is the property.
&prop is the property which gets returned.

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by abma** [comment]
this change very likely introduced https://github.com/FreeRDP/FreeRDP/issues/4096 :

fullscreen mode doesn't work any more.

**Pull request merged**